cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserDAO.java

```diff
@@ -351,6 +353,14 @@ private String getEmailAddress(final User user)
     private void checkUsers(final Principal userID, final String email, final String usersDN)
         throws TransientException, UserAlreadyExistsException
     {
+        // Internal system property set during LdapUserDAOTest.testAddMultipleUser
+        // so that this check can be overridden
+        if (System.getProperty(SUPPRESS_CHECKUSER_KEY) != null && System.getProperty(SUPPRESS_CHECKUSER_KEY).equals("true"))
+        {
```

Comment:
Not a very big fan of this approach. Subclassing LdapUserDAO in the test code and overriding this method would have achieved the same effect without creating this backdoor.

Reply:
I hear you. This was at Pat's suggestion. It was something that isn't supposed to be documented or available publicly. It's used only in testing.

Reply:
Not documented but available publicly. Why is the initial value "cadc.skip.checkuser" and not just null? Does it have a meaning?